Source/Core/Core/HW/EXI/EXI_DeviceMemoryCard.cpp

```diff
@@ -158,10 +158,39 @@ CEXIMemoryCard::CEXIMemoryCard(const int index, bool gciFolder) : card_index(ind
   SetCardFlashID(header.data(), card_index);
 }
 
-void CEXIMemoryCard::SetupGciFolder(u16 sizeMb)
+std::string CEXIMemoryCard::GetGCIFolderPath(int card_index, bool allow_movie_folder, bool* migrate)
```

Comment:
Can this return a `std::pair<std::string, bool>` to avoid the out-param?

Reply:
It could, but then there would be nothing in the function declaration that tells you what the bool means, so I think it's better the way it is right now.

Reply:
That is not necessarily a problem.
`std::pair<std::string /* ... */, bool /* migrate */>` is a common way to define a pair, then you can use structured bindings to unpack the pair on assignment.

Reply:
Just one question about structured bindings before I change this. Calling the modified function like this works:

```
const auto [strDirectoryName, migrate] = GetGCIFolderPath(card_index, true);
```

But is there a way to specify the type of the two variables separately, so that I don't have to use auto? That way you wouldn't have to look up the function to be able to know what types the two variables are.

Reply:
> But is there a way to specify the type of the two variables separately, so that I don't have to use auto?

No :(

Reply:
I've made it use a structured binding.